cmd/collector/app/handler/otlp_receiver_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+)
+
+func TestStartOtlpReceiver(t *testing.T) {
+	spanProcessor := &mockSpanProcessor{}
+	logger, _ := testutils.NewLogger()
+	rec, err := StartOtelReceiver(
+		OtelReceiverOptions{
+			GRPCHostPort: ":0",
+			HTTPHostPort: ":0",
+		},
+		logger,
+		spanProcessor,
+	)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, rec.Shutdown(context.Background()))
+	}()
+
+	// Ideally, we want to test with a real gRPC client, but OTEL repos only have those as internal packages.
+	// So we will rely on otlpreceiver being tested in the OTEL repos, and we only test the consumer function.
+}
+
+func makeTracesOneSpan() ptrace.Traces {
+	traces := ptrace.NewTraces()
+	rSpans := traces.ResourceSpans().AppendEmpty()
+	sSpans := rSpans.ScopeSpans().AppendEmpty()
+	span := sSpans.Spans().AppendEmpty()
+	span.SetName("test")
+	return traces
+}
+
+func TestConsumerDelegate(t *testing.T) {
+	testCases := []struct {
+		expectErr error
+		expectLog string
+	}{
+		{}, // no errors
```

Comment:
I would add a test `name` or `description` so there won't be a need for this comment. i.e.:

```
{name: "no errors"},
...
t.Run(test.name, func(t *testing.T) {...
```

It has the added benefit of a more descriptive test name in logs when executing the test.